meetings/2021-07-28.md

```diff
@@ -0,0 +1,136 @@
+# 2021-07-28 Solid Authorization
+
+https://meet.jit.si/solid-authorization
+
+
+## Agenda
+
+* Minutes - https://github.com/solid/authorization-panel/pull/240
+* UC0 - https://github.com/solid/authorization-panel/pull/244
+* UC3 - https://github.com/solid/authorization-panel/pull/241
+* ACL on ACL - https://github.com/solid/authorization-panel/issues/189
+* Mode range: https://github.com/solid/authorization-panel/issues/187
+
+## Present
+
+* Elf Pavlik
+* Sarven Capadisli
+* Henry Story
+* Matthieu Bosquet
+
+
+## Minutes
+
+## UC0
+
+Sarven: What happens when you get a 403 on ACR?
+
+Matthieu: I don't think we have a good required credentials discovery mechanism. That's what we need defining.
+
+Sarven: You have foo and foo.acl. If I get a 404 on foo.acl, it tells me that foo potentially exists. It emphasises on the authorise. If you know that foo exists but don't have access to it, getting a 403 (or 404) is fine.
+
+Elf: Go to foo, get a header to the acl, how do you know?
+
+Sarven: Whether you know or not that foo.acl exists. It's fine, but if you don't know it exists, what should happen?
+
+Elf: Without knowing that foo exists, you don't get to know that the acl exists.
+
+Sarven: If you don't know that foo exists you should get a 403.
+
+Elf: How do you get a response you don't get a response.
+
+Sarven: Just guessing URLs. If I look at the resource directly, I can see that file exists.
+But let's say I don't have read access to the container but I'm guessing the filename "work-on-the-weekend.txt" and then I'm getting to the acl.
+If you don't know the existence of a resource you get a 403.
+
+Henry: How would the server know what you know?
+
+Sarven: Because you're authorized (or not).
+
+Henry: I think we need to go at it from an unauthenticated perspective.
+The first thing we have to do is how to build apps that work accross pods?
+Everyone is using social networks because it's easy to use.
+If we can get everybody to be working on decentralised apps, we're solving a real problem. Not how does the CIA deals with it?
+For the moment, the problem is how does an app following links knows how to authenticate?
+Currently with WAC, it can't find out.
+Because right now, you follow a link and the only person that follows the link and wants to read the ACL is the controller. There is no other way of saying it.
+If I am a friend of yours and follow links and want to read something, I have no idea what identity to use in order to access resources.
+This is not a decentralised network, it is 1990s PC.
+There are a ton of experiments that put a user at the center of using a cloud based system.
+That is not decentralisation.
+We can't make the same mistake.
+The question is how do I comming accross a link what the access control rules are.
+There is a simple answer, which would be making the ACLs more readable than the resources.
```

Comment:
```suggestion
Henry: I think we need to go at it from an unauthenticated perspective.
The first thing we have to cover is how to build apps that work across pods.
Everyone is using social networks because they're easy to use.
If we can get everybody to be working on decentralised apps, we're solving a real problem. Not how the CIA deals with it.
For the moment, the problem is how an app following links knows how to authenticate?
Currently, with WAC, it can't find out.
Because right now, you follow a link, and the only person that follows the link and wants to read the ACL is the controller. There is no other way of saying it.
If I am a friend of yours and follow links and want to read something, I have no idea what identity to use in order to access resources.
This is not a decentralised network, it is 1990s PC.
There are a ton of experiments that put a user at the center of using a cloud-based system.
That is not decentralisation.
We can't make the same mistake.
The question is, how do I, coming across a link, learn what the access control rules are?
There is a simple answer, which would be making the ACLs more readable than the resources.
```